### PR TITLE
OBSDOCS-1372 - Logging 6.1.0 Release Notes

### PR DESCRIPTION
**Pull request by libander**
Version(s): 4.15, 4.16, 4.17

Issue: [OBSDOCS-1372](https://issues.redhat.com//browse/OBSDOCS-1372) 

Link to docs preview:

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/log6x-6-1-0-rn.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+= Logging 6.1.0 Release Notes
+// Module included in the following assemblies:
+//
+
+:mod-docs-content-type: REFERENCE
+[id="logging-release-notes-6-1-0_{context}"]
```
[the remark this answers is not in the document]
Comment (libander) [reply modules/log6x-6-1-0-rn.adoc:6]: False.

File: modules/log6x-6-1-0-rn.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+* This enhancement adds the source `iostream` to the attributes sent from collected container logs. The value is set to either `stdout` or `stderr` based on how the collector received it. (link:https://issues.redhat.com/browse/LOG-5292[LOG-5292])
+
+* With this update, the default memory limit for the collector increases from 1024Mi to 2024Mi. Users should adjust resource limits based on their cluster’s specific needs and specifications. (link:https://issues.redhat.com/browse/LOG-6072[LOG-6072])
+
+* With this update, users can now set the syslog output delivery mode to either AtLeastOnce or AtMostOnce. (link:https://issues.redhat.com/browse/LOG-6355[LOG-6355])
+
+=== Log Storage
+
+* With this update, the new 1x.extra-small LokiStack size supports OCP clusters with fewer workloads and lower log volumes (up to 50GB/day). (link:https://issues.redhat.com/browse/LOG-5939[LOG-5939])
```
Comment (periklis) [inline modules/log6x-6-1-0-rn.adoc:23]:
```suggestion
* With this update, the new 1x.pico LokiStack size supports OCP clusters with fewer workloads and lower log volumes (up to 50GB/day). (link:https://issues.redhat.com/browse/LOG-5939[LOG-5939])
```

File: modules/log6x-6-1-0-rn.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+[id="logging-release-notes-6-1-0-technology-preview-features"]
+== Technology Preview Features
+
+* This enhancement adds the originating `iostream` to the attributes forwarded from collected container logs. The value will be set to either `stdout` or `stderr` depending on how it was ingested by the collector. (link:https://issues.redhat.com/browse/LOG-5637[LOG-5637])
+
+
```
Comment (periklis) [inline modules/log6x-6-1-0-rn.adoc:30]: Missing the OpenTelemetry end-2-end support as TP Feature here, right?

File: modules/log6x-6-1-0-rn.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+[id="openshift-logging-release-notes-6-1-0-enhancements"]
+== New Features and Enhancements
+
+=== Log Collection
+
+* This enhancement adds the source `iostream` to the attributes sent from collected container logs. The value is set to either `stdout` or `stderr` based on how the collector received it. (link:https://issues.redhat.com/browse/LOG-5292[LOG-5292])
+
+* With this update, the default memory limit for the collector increases from 1024Mi to 2024Mi. Users should adjust resource limits based on their cluster’s specific needs and specifications. (link:https://issues.redhat.com/browse/LOG-6072[LOG-6072])
+
+* With this update, users can now set the syslog output delivery mode to either AtLeastOnce or AtMostOnce. (link:https://issues.redhat.com/browse/LOG-6355[LOG-6355])
```
Comment (xperimental) [inline modules/log6x-6-1-0-rn.adoc:19]: Should we format the values?

```suggestion
* With this update, users can now set the syslog output delivery mode to either `AtLeastOnce` or `AtMostOnce`. (link:https://issues.redhat.com/browse/LOG-6355[LOG-6355])
```

File: modules/log6x-6-1-0-rn.adoc (hunk 2 above)
Comment (xperimental) [reply modules/log6x-6-1-0-rn.adoc:23]: Added formatting to Peri's suggestion:

```suggestion
* With this update, the new `1x.pico` LokiStack size supports OCP clusters with fewer workloads and lower log volumes (up to 50GB/day). (link:https://issues.redhat.com/browse/LOG-5939[LOG-5939])
```

File: modules/log6x-6-1-0-rn.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+=== Log Storage
+
+* With this update, the new 1x.extra-small LokiStack size supports OCP clusters with fewer workloads and lower log volumes (up to 50GB/day). (link:https://issues.redhat.com/browse/LOG-5939[LOG-5939])
+
+[id="logging-release-notes-6-1-0-technology-preview-features"]
+== Technology Preview Features
+
+* This enhancement adds the originating `iostream` to the attributes forwarded from collected container logs. The value will be set to either `stdout` or `stderr` depending on how it was ingested by the collector. (link:https://issues.redhat.com/browse/LOG-5637[LOG-5637])
```
Comment (xperimental) [inline modules/log6x-6-1-0-rn.adoc:28]: If this is a tech-preview feature, should this be made clear in the "new features" section above as well? Another option would be to just list it here.

**Review by xperimental — COMMENTED** [review]
Added a few comments.

Brian, Satya and I also had a short meeting about some of the structure of the Logging 6.1 documentation where I said, that for me it doesn't make sense to have the full "Upgrading to 6.0" document branded as "Upgrading to 6.1" again in the documentation.

Does it make sense to instead link to that Upgrade document from the release notes stating something like "if you're upgrading from a version before 6.0, please follow the steps outlined in the Upgrading to 6.0 document first"?

File: modules/log6x-6-1-0-rn.adoc (hunk 5 above)
Comment (jcantrill) [inline modules/log6x-6-1-0-rn.adoc:28]: This description is incorrect as the link as for e2e OTLP support an not the iostream feature which was properly listed  as an enhancement

Comment (libander) [reply modules/log6x-6-1-0-rn.adoc:28]: No, it gets its own section. https://redhat-documentation.github.io/supplementary-style-guide/#technology-preview-guidance

Comment (libander) [reply modules/log6x-6-1-0-rn.adoc:28]: Nah, tech previews have their own rules per the supplemental style guide.

File: observability/logging/logging-6.0/log6x-clf.adoc [hunk 6]
```diff
@@ -88,6 +88,8 @@ syslog:: Forwards logs to an external syslog server.
 
 Each output type has its own configuration fields.
 
+include::modules/log6x-configuring-otlp-output.adoc[leveloffset=+1]
```
Comment (xperimental) [inline observability/logging/logging-6.0/log6x-clf.adoc:91]: Was this line and the "configuring otlp output" file added to this PR intentionally? It seems unrelated to the release notes and the pico-size PR also seems to include the same file.

**Review by xperimental — COMMENTED** [review]
Is this PR missing an update? Some of the lines with resolved comments have not changed.

File: observability/logging/logging-6.0/log6x-clf.adoc (hunk 6 above)
Comment (libander) [reply observability/logging/logging-6.0/log6x-clf.adoc:91]: nope, its a release branch thing, it isnt actually part of this PR.

Comment (xperimental) [reply observability/logging/logging-6.0/log6x-clf.adoc:91]: I'm not sure I understand the answer. It looks to me as if the first commit of this PR should not be in it. Is that what you are saying?

File: modules/log6x-6-1-0-rn.adoc [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+For more information about Technology Preview support, see the link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope].
+====
+
+=== Configuration
+
+Specify `otlp` as the output type in the `ClusterLogForwarder` to enable OTLP output. Configuration options include:
+
+- Setting the `url` for the OTLP receiver endpoint.
+- Configuring `tls` settings with `insecureSkipVerify`, if required.
+- Setting `authentication` with a service account token.
```
[the remark this answers is not in the document]
Comment (cahartma) [reply modules/log6x-6-1-0-rn.adoc:43]: I agree.   These are the incorrect steps to forward otlp logs to lokistack.   Instead this needs to reference the new `lokiStack.dataModel` field, and setting the value to `Otel`.
Here's our upstream feature doc as well:    https://github.com/openshift/cluster-logging-operator/blob/master/docs/features/logforwarding/outputs/opentelemetry-lokistack-forwarding.adoc

File: modules/log6x-6-1-0-rn.adoc [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+* With this update, users can now set the syslog output delivery mode to either `AtLeastOnce` or `AtMostOnce.` (link:https://issues.redhat.com/browse/LOG-6355[LOG-6355])
+
+=== Log Storage
+
+* With this update, the new `1x.pico` LokiStack size supports clusters with fewer workloads and lower log volumes (up to 50GB/day). (link:https://issues.redhat.com/browse/LOG-5939[LOG-5939])
+
+[id="logging-release-notes-6-1-0-technology-preview-features"]
+== Technology Preview
+
+With this update, the `ClusterLogForwarder` resource now supports forwarding logs to OpenTelemetry-compatible receivers using HTTP/JSON, as defined by the OpenTelemetry OTLP specification. This enhancement allows logs to be transported to Red Hat Managed Lokistack or other OTLP-compliant services.
```
Comment (cahartma) [inline modules/log6x-6-1-0-rn.adoc:28]: This specific release is only related to sending to lokiStack, although the "tech-preview" relates to anything OTLP at this point.   We previously released the otlp (generic) output in 6.0, which sends to compatible receivers.   For 6.1, we are adding the feature to the lokiStack output spec, via the new `dataModel` field.

Comment (cahartma) [reply modules/log6x-6-1-0-rn.adoc:28]: Additionally, or maybe not worth the mention, but our lokiStack now "accepts" the collector's otel logs, which is the other part of the feature (along with query and UI work).

File: modules/log6x-6-1-0-rn.adoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+This release includes link:https://access.redhat.com/errata/RHBA-2024:9038[{OCP-short}{logging} Release 6.1.0].
+
+[id="openshift-logging-release-notes-6-1-0-enhancements"]
+== New Features and Enhancements
+
+=== Log Collection
+
+* This enhancement adds the source `iostream` to the attributes sent from collected container logs. The value is set to either `stdout` or `stderr` based on how the collector received it. (link:https://issues.redhat.com/browse/LOG-5292[LOG-5292])
+
+* With this update, the default memory limit for the collector increases from 1024Mi to 2024Mi. Users should adjust resource limits based on their cluster’s specific needs and specifications. (link:https://issues.redhat.com/browse/LOG-6072[LOG-6072])
```
[the remark this answers is not in the document]
Comment (briandooley) [reply modules/log6x-6-1-0-rn.adoc:17]: This will be covered in https://issues.redhat.com/browse/OBSDOCS-1494

File: modules/log6x-6-1-0-rn.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (briandooley) [reply modules/log6x-6-1-0-rn.adoc:43]: This will be covered in https://issues.redhat.com/browse/OBSDOCS-1494

File: modules/log6x-6-1-0-rn.adoc (hunk 8 above)
Comment (briandooley) [reply modules/log6x-6-1-0-rn.adoc:28]: This will be covered in https://issues.redhat.com/browse/OBSDOCS-1494

**Comment by briandooley** [comment]
@anpingli - lgtm please.

**Comment by anpingli** [comment]
LGTM

**Comment by mburke5678** [comment]
@libander I do not see a preview here. I am aware that there is a separate [PR](https://github.com/openshift/openshift-docs/pull/84861) to add 6.1 docs, but the 6.1 RN are not present.

File: modules/log6x-6-1-0-rn.adoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
+= Logging 6.1.0 Release Notes
```
Comment (skrthomas) [inline modules/log6x-6-1-0-rn.adoc:1]: @libander I'm merge reviewing the Differntiation [PR](https://github.com/openshift/openshift-docs/pull/84861/files#diff-2620632a1c41ceaaaa2cc8438f93892424205fcd52cdab1d11cf904bbf08bff0), and I am wondering if the 6.1 RN file there, which is titled 6.0.0, and this file(https://github.com/openshift/openshift-docs/pull/84861/files#diff-2620632a1c41ceaaaa2cc8438f93892424205fcd52cdab1d11cf904bbf08bff0) will clash?

Comment (skrthomas) [reply modules/log6x-6-1-0-rn.adoc:1]: cc @smunje1 @briandooley

**Review by xperimental — APPROVED** [review]
Clicking approve, so that the content can be merged into the branch and the docs for 6.1 can be finalized.

I understand that the comments have all be moved to a future cleanup. My suggestion would be to at least fix the typo in the collector memory limit.

File: modules/log6x-6-1-0-rn.adoc [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+[id="openshift-logging-release-notes-6-1-0-enhancements"]
+== New Features and Enhancements
+
+=== Log Collection
+
+* This enhancement adds the source `iostream` to the attributes sent from collected container logs. The value is set to either `stdout` or `stderr` based on how the collector received it. (link:https://issues.redhat.com/browse/LOG-5292[LOG-5292])
+
+* With this update, the default memory limit for the collector increases from 1024Mi to 2024Mi. Users should adjust resource limits based on their cluster’s specific needs and specifications. (link:https://issues.redhat.com/browse/LOG-6072[LOG-6072])
+
+* With this update, users can now set the syslog output delivery mode to either `AtLeastOnce` or `AtMostOnce.` (link:https://issues.redhat.com/browse/LOG-6355[LOG-6355])
```
Comment (mburke5678) [inline modules/log6x-6-1-0-rn.adoc:19]: Do we need to show users how to set this?

**Comment by mburke5678** [comment]
@libander A couple of comments. There is no preview available; I cannot confirm that the links work (I did copy/paste the URLs and they all worked). Also, I cannot confirm that the module is included in an assembly. There is a [Logging 6.x reorg PR](https://github.com/openshift/openshift-docs/pull/84861); perhaps the module will be added there?

File: modules/log6x-6-1-0-rn.adoc (hunk 11 above)
Comment (libander) [reply modules/log6x-6-1-0-rn.adoc]: Added where those fields are.

File: modules/log6x-6-1-0-rn.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (libander) [reply modules/log6x-6-1-0-rn.adoc]: fixed.

File: observability/logging/logging-6.1/log6x-about-6.1.adoc [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+OpenShift Logging supports two data models:
+
+* ViaQ (General Availability)
+* OpenTelemetry (Technology Preview)
+
+You can select either of these data models based on your requirement by configuring the `lokiStack.dataModel` field in the `ClusterLogForwarder`. ViaQ is the default data model when forwarding logs to LokiStack. For more information on OpenTelemetry, see xref:../../../observability/logging/logging-6.1/log6x-opentelemetry-data-model-6.1.adoc#log6x-opentelemetry-data-model-6.1[OpenTelemetry data model].
+
+[NOTE]
+====
+In future releases of OpenShift Logging, the default data model will change from ViaQ to OpenTelemetry.
```
Comment (libander) [inline observability/logging/logging-6.1/log6x-about-6.1.adoc:53]: > In future releases of OpenShift Logging, the default data model will change from ViaQ to OpenTelemetry.

This is:
- not an approved short name for logging 
- [explicitly forbidden by the supplementary style guide. ](https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future)

File: observability/logging/logging-6.1/log6x-about-6.1.adoc [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+- When set to `Managed` (default), the operator actively manages the logging resources to match the configuration defined in the spec.
+- When set to `Unmanaged`, the operator does not take any action, allowing you to manually manage the logging components.
+
+== Validation
+Logging includes extensive validation rules and default values to ensure a smooth and error-free configuration experience. The `ClusterLogForwarder` resource enforces validation checks on required fields, dependencies between fields, and the format of input values. Default values are provided for certain fields, reducing the need for explicit configuration in common scenarios.
+
+[id="quick-start_{context}"]
+== Quick start
+
+OpenShift Logging supports two data models:
```
Comment (libander) [inline observability/logging/logging-6.1/log6x-about-6.1.adoc:44]: Not an approved short name for logging.

File: observability/logging/logging-6.1/log6x-configuring-lokistack-otlp-6.1.adoc [hunk 14]
```diff
@@ -0,0 +1,151 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="log6x-configuring-lokistack-otlp-6.1"]
+= OTLP data ingestion in Loki
+include::_attributes/common-attributes.adoc[]
+:context: log6x-configuring-lokistack-otlp-6.1
+
+toc::[]
+
+Loki 3.0 introduces an API endpoint using the OpenTelemetry Protocol (OTLP), providing a new way to ingest log entries into Loki. This endpoint complements the standard Push API, which is available in Loki from its initial release. Because OTLP is a standardized format not specifically designed for Loki, it requires additional configuration on Loki's side to map OpenTelemetry's data format to Loki's data model. OTLP lacks concepts such as _stream labels_ or _structured metadata_. Instead, OTLP provides metadata about log entries as *attributes*, grouped into three categories:
```
Comment (libander) [inline observability/logging/logging-6.1/log6x-configuring-lokistack-otlp-6.1.adoc]: > Loki 3.0 introduces

We have never referred to the version of Loki before.
